org.adempiere.base/src/org/compiere/model/X_M_Product.java

```diff
@@ -27,6 +27,7 @@
 /** Generated Model for M_Product
  *  @author iDempiere (generated) 
  *  @version Release 8.2 - $Id$ */
+@org.adempiere.base.Model(table="M_Product", intermediate = true)
```

Comment:
if needed, we should run model generator for all existing classes, not just 1 single class here. I understand that this served as an example but this is not appropriate for a PR.

Reply:
We didn't worry much about this, as accepting this PR might imply that all X classes might get annotated. Though it isn't required, we'd like that core model classes get detected and loaded by using this method instead of  the current ```Class.forName``` based approach. However, we've reverted this commit for now.